From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shane Freeder <theboyetronic@gmail.com>
Date: Fri, 18 Oct 2019 17:39:05 +0100
Subject: [PATCH] Set true custom payload channel size limit

This fixes compatibility with some mods that are sending very long channel names. Also gives developers the ability to send logner channel names.

diff --git a/src/main/java/org/bukkit/plugin/messaging/Messenger.java b/src/main/java/org/bukkit/plugin/messaging/Messenger.java
index 9d2c68c826f3b867d407e7f13c6394a899cc8ee8..ca6442ebe6a01a546e2f6d5c79583189ed0b0292 100644
--- a/src/main/java/org/bukkit/plugin/messaging/Messenger.java
+++ b/src/main/java/org/bukkit/plugin/messaging/Messenger.java
@@ -24,7 +24,7 @@ public interface Messenger {
     /**
      * Represents the largest size that a Plugin Channel may be.
      */
-    public static final int MAX_CHANNEL_SIZE = 64;
+    public static final int MAX_CHANNEL_SIZE = 32767; // Paper - set true max channel size
 
     /**
      * Checks if the specified channel is a reserved name.
